t/bib_rights.t

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+BEGIN {
+  unshift(@INC, $ENV{'SDRROOT'}. '/crms/cgi');
+  unshift(@INC, $ENV{'SDRROOT'}. '/crms/post_zephir_processing');
+}
+
+use utf8;
+binmode(STDOUT, ':encoding(UTF-8)');
+use CRMS;
+#use Getopt::Long qw(:config no_ignore_case bundling);
+use Encode;
+#use Term::ANSIColor qw(:constants colored);
+#$Term::ANSIColor::AUTORESET = 1;
+use Data::Dumper;
+use JSON::XS;
+use bib_rights;
+use MARC::File::XML(BinaryEncoding => 'utf8');
+use Test::More;
+
+my $crms = CRMS->new();
+my $fixtures_dir = $ENV{'SDRROOT'} . '/crms/t/fixtures/bib_rights/';
+my $test_struct = read_json('bib_rights_tests.json');
+  
+foreach my $htid (sort keys %$test_struct) {
+  run_tests_for_htid($htid);
+}
+
+### MISCELLANEOUS COVERAGE TESTS
+delete $ENV{BIB_RIGHTS_DATE};
+
+# FIXME: this is a hacked-together example that should be replaced with a real one.
+# Ideally we would have the full OCLC list available in its host submodule for CRMS when it does
+# the /htapps/babel/archive/catalog dump to crms.bib_rights_bi and crms.bib_rights_bri.
+$ENV{'us_fed_pub_exception_file'} = $fixtures_dir . 'feddoc_oclc_nums.txt';
+my $fake_marc_xml = <<END_OF_RECORD;
+<record>
+  <leader>03895cas a2200517I  4500</leader>
+  <controlfield tag="001">000000000</controlfield>
+  <controlfield tag="008">720225m19679999dcu      b   f000 0 eng u</controlfield>
+  <datafield tag="035" ind1=" " ind2=" "><subfield code="a">(OCoLC)000</subfield></datafield>
+</record>
+END_OF_RECORD
+my $fake_marc = MARC::Record->new_from_xml($fake_marc_xml);
+my $br = bib_rights->new();
+my $bib_info = $br->get_bib_info($fake_marc, 'BOGUS RECORD 1');
+ok($bib_info->{us_fed_pub_exception} && $bib_info->{us_fed_pub_exception} eq 'exception list');
+
+# Pass nonexistent fed docs file
+$ENV{'us_fed_pub_exception_file'} = $fixtures_dir . 'no_such_file.txt';
+$br = bib_rights->new();
+# Clone existing BR object
+ok(my $br2 = $br->new());
+
+delete $ENV{'us_fed_pub_exception_file'};
+
+my $date = bib_rights::getDate(0);
+is($date, '19700101:00:00:00');
+
+$fake_marc_xml = '<record><leader>03895cas a2200517I  4500</leader></record>';
+$fake_marc = MARC::Record->new_from_xml($fake_marc_xml);
+$bib_info = $br->get_bib_info($fake_marc, 'BOGUS RECORD 2');
+ok(!scalar keys %$bib_info);
+
+$fake_marc_xml = '<record></record>';
+$fake_marc = MARC::Record->new_from_xml($fake_marc_xml);
+$bib_info = $br->get_bib_info($fake_marc, 'BOGUS RECORD 3');
+ok(!scalar keys %$bib_info);
+
```

Comment:
Would be helpful to add comments on which particular edge cases are being tested.